DQM/SiPixelHeterogeneous/plugins/SiPixelCompareTrackSoAAlpaka.cc

```diff
@@ -0,0 +1,318 @@
+// -*- C++ -*-
+// Package:    SiPixelCompareTrackSoAAlpaka
+// Class:      SiPixelCompareTrackSoAAlpaka
+//
+/**\class SiPixelCompareTrackSoAAlpaka SiPixelCompareTrackSoAAlpaka.cc
+*/
+//
+// Author: Suvankar Roy Chowdhury
+//
+
+// for string manipulations
+#include <fmt/printf.h>
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+// DQM Histograming
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+// DataFormats
+#include "DataFormats/TrackSoA/interface/TracksHost.h"
+#include "DataFormats/TrackSoA/interface/alpaka/TrackUtilities.h"
+
+namespace {
+  // same logic used for the MTV:
+  // cf https://github.com/cms-sw/cmssw/blob/master/Validation/RecoTrack/src/MTVHistoProducerAlgoForTracker.cc
```

Comment:
How about moving the MTV functions to some common place to avoid code duplication (at least in "near future")?

Reply:
I agree with this and would add it to the TODO issue we will open. Just because this means touching few other packages I would keep separate from this PR.

Reply:
As this block is a straight copy from `SiPixelCompareTrackSoA` (which is in the same package)

https://github.com/cms-sw/cmssw/blob/66176a172662e116a37a6c68c80974d95b0a2910/DQM/SiPixelHeterogeneous/plugins/SiPixelCompareTrackSoA.cc#L28-L64

perhaps can be moved into a "utility" header file in the same package.